Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -0,0 +1,199 @@
+            if self.dialect == "Microsoft SQL Server":
+                db_preferences += "?driver=FreeTDS"
+            if self.connect_parameters:
+                db_preferences += f'?{self.connect_parameters}'
+            return sqlalchemy.create_engine(db_preferences).connect()
+        except Exception as err:
+            raise Exception(err)
+
+    def sql_query_execute_request(self, sql_query: str, bind_vars: Any) -> Tuple[Dict, List]:
+        """Execute query in DB via engine
+        :param bind_vars: in case there are names and values - a bind_var dict, in case there are only values - list
+        :param sql_query: the SQL query
+        :return: results of query, table headers
+        """
+        if type(bind_vars) is dict:
+            sql_query = text(sql_query)
+
+        result = self.connection.execute(sql_query, bind_vars)
+        results = result.fetchall()
+        headers = results[0].keys()
+        return results, headers
+
+
+def generate_default_port_by_dialect(dialect: str) -> str:
+    """
+    In case no port wash chosen, a default port will be chosen according to the SQL db type
+    :param dialect: sql db type
+    :return: default port needed for connection
+    """
+    if dialect == "MySQL":
+        return "3306"
+    elif dialect == "PostgreSQL":
+        return "5432"
+    elif dialect == "Oracle":
+        return "1521"
+    elif dialect == "Microsoft SQL Server":
+        return "1433"
+    else:
+        # set default to mysql
+        return "3306"
+
+
+def generate_bind_vars(bind_variables_names: str, bind_variables_values: str) -> Any:
+    """
+    The bind variables can be given in 2 legal ways: as 2 lists - names and values, or only values
+    any way defines a different executing way, therefore there are 2 legal return types
+    :param bind_variables_names: the names of the bind variables, must be in the length of the values list
+    :param bind_variables_values: the values of the bind variables, can be in the length of the names list
+            or in case there is no name lists - at any length
+    :return: a dict or lists of the bind variables
+    """
+    bind_variables_names_list = argToList(bind_variables_names)
+    bind_variables_values_list = argToList(bind_variables_values)
+
+    if bind_variables_values and not bind_variables_names:
+        return [var for var in argToList(bind_variables_values)]
+    elif len(bind_variables_names_list) is len(bind_variables_values_list):
+        return dict(zip(bind_variables_names_list, bind_variables_values_list))
+    else:
+        raise Exception("The bind variables lists are not is the same length")
+
+
+def test_module(client: Client, *_) -> Tuple[str, Dict[Any, Any], List[Any]]:
+    """
+    Performs basic connect to SQL server
+    If the connection in the client was successful the test will return OK
+    if it wasn't an exception will be raised
```

Comment:
this function does not perform the connection, it was already tested in the the Client. please update the docstring.